### PR TITLE
Feat/openapi docs

### PR DESCRIPTION
**Pull request by taavettik**
Replace the custom made api documentation generation method with OpenAPI-compatible schema generation from router JOI types

### CODE REVIEW

**Comment by keskiju** [comment]
Looks good 👍 

Currently on rest.ts we have:

```

**Comment by keskiju** [comment]
Looks good 👍 

On rest.ts we have:

```
if (config.COMMON_ENV !== 'prod') {
  restMiddlewares.push(apiDocRouter.middleware());
}
```

And on graphql.ts we have:

```
  introspection: ['local', 'dev', 'test'].includes(config.COMMON_ENV || ''),
  playground: ['local', 'dev', 'test'].includes(config.COMMON_ENV || '')
```

We should unify these. `[ 'local' ]` would be a safe default as this way you don't make docs/introspection and playground public by accident, but you need to intentionally enable them for each environment. Maybe we should have environment variables `API_DOCS_ENABLED: true` and `API_PLAYGROUND_ENABLED: true`; enabled on docker-compose.yaml by default, and on helm-dev.yaml like this:

```
full-stack:
  services:
    server:
      env:
        # Enable if API docs and playground required on dev.
        # API_DOCS_ENABLED: true
        # API_PLAYGROUND_ENABLED: true
```

**Review by keskiju — CHANGES_REQUESTED** [review]
PR ok, but see the aforementioned comment.

**Comment by taavettik** [comment]
> Maybe we should have environment variables `API_DOCS_ENABLED: true` and `API_PLAYGROUND_ENABLED: true`; enabled on docker-compose.yaml by default, and on helm-dev.yaml like this:

I'd say that sounds wise. Will do

**Review by keskiju — CHANGES_REQUESTED** [review]

File: server/src/common/setup/config.ts [hunk 1]
```diff
@@ -85,6 +85,8 @@ const config = {
     : 4000,
   API_BINDADDR: process.env.API_BINDADDR || '127.0.0.1',
   BASE_PATH: process.env.BASE_PATH || '/api',
+  API_DOCS_ENABLED: process.env.API_DOCS_ENABLED?.toString() === 'true',
+  API_PLAYGROUND_ENABLED: Boolean(process.env.API_PLAYGROUND_ENABLED),
```
[the remark this answers is not in the document]
Comment (taavettik) [reply server/src/common/setup/config.ts:89]: brainfart

**Pull request merged**